packages/@react-aria/selection/src/useSelectableCollection.ts

```diff
@@ -364,6 +377,20 @@ export function useSelectableCollection(options: SelectableCollectionOptions): S
     }
   }, [isVirtualized, scrollRef, manager.focusedKey]);
 
+  // Update the focused key if the collection isn't focused and the selected keys changes to something that doesn't contain the focused key.
+  // Don't do this for virtual focus collections because the user is never focused within the collection and thus we don't
+  // want to mistakenly change their focus position.
+  let lastSelectedKeys = useRef(manager.selectedKeys);
+  useEffect(() => {
+    if (!equalSets(lastSelectedKeys.current, manager.selectedKeys)) {
+      if (!manager.isSelected(manager.focusedKey) && !manager.isFocused && !shouldUseVirtualFocus) {
+        manager.firstSelectedKey != null && manager.setFocusedKey(manager.firstSelectedKey);
+      }
+    }
+
+    lastSelectedKeys.current = manager.selectedKeys;
+  }, [manager.focusedKey, manager.isFocused, manager.firstSelectedKey, manager.selectedKeys, shouldUseVirtualFocus]);
```

Comment:
IMO, it felt weird to be modifying the focusedKey directly in the stately hook, didn't feel like it was the right place to be handling focus manager. Additionally, the stately hook doesn't have access to the selectionManager/collection so if the `selectedKeys` was set to "all", we wouldn't be able to look up what the first key actually is and set `focusedKey` appropriately

Reply:
Updated to remove the `useEffect`. Note that this caused "setState in render" warnings that I've tentatively addressed, lemme know what you think.

Reply:
hmmm but we track the focusedKey in state? so seems like we can update it to whatever in the state. It's not like it's happening as the result of a user interaction. It's happening because of other state updating

Reply:
Fair enough. We still have the issue of not having access to the collection within `useMultipleSelectionState` so we can't look up what the first key is if `selectedKeys` is set to `all` and thus can't properly set what key to change `focusedKey` to. 

Should we have `collection` be passed to `useMultipleSelectionState` (I don't like this option since it adds a potentially required prop for this small change in behavior). Another option is to limit this behavior to `useSingleSelectListState`?

Reply:
seems like useMultipleSelectionState should have access to the collection if the other hooks do? annoying that it would probably be a required prop, but maybe we can just do what we do today if the collection isn't passed in? make it optional but issue a deprecation if we don't get it? good news is that these are still mostly internal only?